assets/js/pages/Users/UsersPage.jsx

```diff
@@ -0,0 +1,69 @@
+import React, { useState, useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { listUsers, deleteUser } from '@lib/api/users';
+
+import { toast } from 'react-hot-toast';
+
+import Users from './Users';
+
+export const SUCCESS_DELETE_MESSAGE = 'User deleted successfully';
+export const ERROR_DELETE_MESSAGE = 'An error occurred: User not found';
+export const ERROR_FETCH_MESSAGE = 'An error occurred: Fetching users failed';
+
+function UsersPage() {
+  const [loading, setLoading] = useState(true);
+  const [users, setUsers] = useState([]);
+  const [error, setError] = useState(null);
+
+  const navigate = useNavigate();
+
+  const fetchUsers = () => {
+    setLoading(true);
+    listUsers()
+      .then((response) => {
+        setUsers(response.data);
+      })
+      .catch((_error) => {
+        setError(ERROR_FETCH_MESSAGE);
+        setUsers([]);
+      })
+      .finally(() => {
+        setLoading(false);
+      });
+  };
+
+  const onDeleteUser = (userID) => {
+    deleteUser(userID)
+      .then(() => {
+        toast.success(SUCCESS_DELETE_MESSAGE);
+      })
+      .catch((_error) => {
+        setError(ERROR_DELETE_MESSAGE);
```

Comment:
By now, this could be alright, but I would be careful of catching all the errors as they were `user not found`.
You could have the forbidden error as well.
Any case, i don't think it can happen in reallity

Reply:
I get your point, we could validate the error message and display different error messages, but I also think that currently only the not found error is possible. 
Maybe we can come back to it later when we add user permissions.
What do you think ?